native/effektio/src/api/room.rs

```diff
@@ -194,6 +206,190 @@ impl Room {
+            })
+            .await?
+    }
+
+    pub async fn send_file_message(
+        &self,
+        uri: String,
+        name: String,
+        mimetype: String,
+        size: u32,
+    ) -> Result<String> {
+        let room = if let MatrixRoom::Joined(r) = &self.room {
+            r.clone()
+        } else {
+            bail!("Can't send message to a room we are not in")
+        };
+        RUNTIME
+            .spawn(async move {
+                let path = PathBuf::from(uri);
+                let mut image = File::open(path)?;
+                let config = AttachmentConfig::new().info(AttachmentInfo::File(BaseFileInfo {
+                    size: Some(UInt::from(size)),
+                }));
+                let mime_type: mime::Mime = mimetype.parse().unwrap();
+                let r = room
+                    .send_attachment(name.as_str(), &mime_type, &mut image, config)
+                    .await?;
+                Ok(r.event_id.to_string())
+            })
+            .await?
+    }
+
+    pub async fn save_file(&self, event_id: String, dir_path: String) -> Result<bool> {
+        let room = if let MatrixRoom::Joined(r) = &self.room {
+            r.clone()
+        } else {
+            bail!("Can't send message to a room we are not in")
+        };
+        let client = self.client.clone();
+        // any variable in self can't be called directly in spawn
+        RUNTIME
+            .spawn(async move {
+                let eid = EventId::parse(event_id.clone())?;
+                let evt = room.event(&eid).await?;
+                match evt.event.deserialize() {
+                    Ok(AnyRoomEvent::MessageLike(AnyMessageLikeEvent::RoomMessage(
+                        MessageLikeEvent::Original(m),
+                    ))) => match m.content.msgtype {
+                        MessageType::File(content) => {
+                            let source = content.source.clone();
+                            let name = content.body.clone();
+                            let mut path = PathBuf::from(dir_path.clone());
+                            path.push(name);
+                            let mut file = File::create(path.clone())?;
+                            let data = client
+                                .get_media_content(
+                                    &MediaRequest {
+                                        source,
+                                        format: MediaFormat::File,
+                                    },
+                                    false,
+                                )
+                                .await?;
+                            file.write_all(&data)?;
+                            let key =
+                                [room.room_id().as_str().as_bytes(), event_id.as_bytes()].concat();
+                            client
+                                .store()
+                                .set_custom_value(&key, path.to_str().unwrap().as_bytes().to_vec())
```

Comment:
please don't `unwrap` in actual code (tests are fine). Instead always use `.expect(MSG)` with a message stating, why we can be sure this never panics. e.g. : 

```suggestion
                                .set_custom_value(&key, path.to_str().expect("Path was generated from strings. Must be string").as_bytes().to_vec())
```

Reply:
Will fix